smsjwplatform/acl.py

```diff
@@ -0,0 +1,182 @@
+LOOKUP_SESSION = AuthenticatedSession(scopes=settings.SMS_OAUTH2_LOOKUP_SCOPES)
+
+
+def get_person_for_user(user):
+    """
+    Return the resource from Lookup associated with the specified user. A requests package
+    :py:class:`HTTPError` is raised if the request fails.
+
+    The result of this function call is cached based on the username so it is safe to call this
+    multiple times.
+
+    If user is the anonymous user (user.is_anonymous is True), :py:class:`~.UserIsAnonymousError`
+    is raised.
+
+    """
+    # check that the user is not anonymous
+    if user.is_anonymous:
+        raise RuntimeError('User is anonymous')
+
+    # return a cached response if we have it
+    cached_resource = cache.get(f"{user.username}:lookup")
+    if cached_resource is not None:
+        return cached_resource
+
+    # Ask lookup about this person
+    lookup_response = LOOKUP_SESSION.request(
+        method='GET', url=urljoin(
+            settings.LOOKUP_ROOT,
+            f'people/{settings.LOOKUP_PEOPLE_ID_SCHEME}/{user.username}?fetch=all_insts,all_groups'
+        )
+    )
+
+    # Raise if there was an error
+    lookup_response.raise_for_status()
+
+    # save cached value
+    cache.set(f"{user.username}:lookup", lookup_response.json(),
+              settings.LOOKUP_PEOPLE_CACHE_LIFETIME)
+
+    # recurse, which should now retrieve the value from the cache
+    return get_person_for_user(user)
+
+
+class AceWorld:
+    """This class encapsulates an ACE of the form WORLD"""
+
+    @staticmethod
+    def parse(ace):
+        """Parses an ACE and constructs/return an AceWorld if it matches or None if it doesn't"""
+        return AceWorld() if ace == 'WORLD' else None
+
+    @staticmethod
+    def has_permission(user):
+        """Anyone can see the media"""
+        LOG.debug(user)
+        return True
+
+
+class AceCam:
+    """This class encapsulates an ACE of the form CAM"""
+
+    @staticmethod
+    def parse(ace):
+        """Parses an ACE and constructs/return an AceCam if it matches or None if it doesn't"""
+        return AceCam() if ace == 'CAM' else None
+
+    @staticmethod
+    def has_permission(user):
+        """Any authenticated user can see the media"""
+        return not user.is_anonymous
+
+
+class AceInst:
+    """This class encapsulates an ACE of the form INST_{instid}"""
+
+    prefix = 'INST_'
+
+    def __init__(self, instid):
+        self.instid = instid
+
+    @staticmethod
+    def parse(ace):
+        """Parses an ACE and constructs/return an AceInst if it matches or None if it doesn't"""
+        return AceInst(ace[len(AceInst.prefix):]) if ace.startswith(AceInst.prefix) else None
+
+    def has_permission(self, user):
+        """Only a user belonging to the 'instid' lookup institution can see the media"""
+        if user.is_anonymous:
+            return False
+
+        lookup_response = get_person_for_user(user)
+
+        for institution in lookup_response['institutions']:
```

Comment:
In line with the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle) maybe:
```python
for institution in lookup_response.get('institutions', []):
    if self.instid == institution.get('instid', None):
        return True
return False
```
or, if you think it more explicit:
```python
return self.instid is not None and any(
    self.instid == institution.get('instid', None)
    for institution in lookup_response.get('institutions', [])
)
```